### ISSUE TITLE
PnP Provisioning Logging / Error Handling

### ISSUE THREAD

**Issue by mrpullen** [SharePoint/PnP-Sites-Core#2055]
Thank you for reporting an issue or suggesting an enhancement. We appreciate your feedback - to help the team to understand your needs, please complete the below template to ensure we have the necessary details to assist you. If you have a actual question, we would ask you to use [SharePoint Developer Group](http://aka.ms/sppnp-community) at Microsoft Tech Community. Thanks!

#### Which PnP repository should you use to report the issue? 
*Please navigate to the appropriate repository by clicking on the link below and create your issue there. You can delete this section after you have navigated to the correct repository.*
- PnP Sites Core - https://github.com/OfficeDev/PnP-Sites-Core
- PnP PowerShell - https://github.com/OfficeDev/PnP-PowerShell

#### Category
[ ] Bug
[X] Enhancement

#### Environment
[X] Office 365 / SharePoint Online
[X] SharePoint 2016
[X] SharePoint 2013

If SharePoint on-premises, what's exact CU version: 

#### Expected or Desired Behavior

The error messages that are returned after a failure when attempting to provision a site using the Apply-PnPProvisioningTemplate need to be improved. There isn't enough information oftentimes to understand clearly where the error is. Additionally in the case of server errors more friendly feedback would make it significantly simpler to troubleshoot and use. Additionally I was able to deploy corrupted site fields which made all site columns unavailable on the site.

I would like to see an output similar to this after the pnp provisioning engine (regardless of success / failure) 

```
SiteFields (19/20 succeeded)
   [X] HideSideNavigation - failed (Unable to provison -- error message / reason)
   [X] PageLookup - failed (error with Pages List not attempting to run)
Content Types (4/5)
   [X] Intranet Page - failed (Site Field HideSideNavigation was not found)
Lists (7/8)
   [X] Pages - failed (Intranet Page Content Type was not available)
...


Provisioning Failed (All Errors will be stored to the $Error powershell field).
```

This would go a long way to making the issues with templates easier to find and resolve.

#### Observed Behavior
_If you are reporting a bug, please describe the behavior you expected to occur when performing the action. If you are making a suggestion, you can delete this section._

The error messages that the Apply-PnPProvisioningTemplate cmdlet produces aren't useful for the most part. 9 times out of 10 I start cutting fields / lists / content types / etc out of the xml I am attempting to deploy until I identify the issue.

**Comment by heinrich-ulbricht** [comment]
Yes, there is lots of room for improvement. The error messages need to state clearly what step exactly failed.
When getting a `ServerUnauthorizedAccessException` I want to know which permissions were missing. Parts of the template have been applied and now I have to manually dig into the template to figure out what might be the issue.

I wished the error message was like "Provisioning term sets failed: access denied".
